components/mpas-framework/src/framework/mpas_io_streams.F

```diff
@@ -3641,30 +3644,50 @@ subroutine MPAS_writeStream(stream, frame, ierr)
                      else
                         call mpas_pool_get_dimension(field_2dreal_ptr % block % dimensions, field_2dreal_ptr % dimNames(2), ownedSize)
                      end if
+                     write(log_string,*) 'i,OwnedSize,dimName = ',i, ownedSize, trim(field_2dreal_ptr % dimNames(2))
+                     call mpas_log_write(log_string)
 
                      if (field_cursor % real2dField % isVarArray) then
+                        call mpas_log_write('... isVarArray decomposed, assigning real1d_temp')
                         real1d_temp(i:i+ownedSize-1) = field_2dreal_ptr % array(j,1:ownedSize)
                      else
-                        real2d_temp(:,i:i+ownedSize-1) = field_2dreal_ptr % array(:,1:ownedSize)
+                        call mpas_log_write('... isVar decomposed, assigning real2d_temp')
+                        !TODO
+                        !if (ASSOCIATED( field_2dreal_ptr % missingValueMask1D )) then
+                        !   call mpas_log_write('masking 2d array with 1d array not yet supported')
+                           !call mpas_mask_field3d_mask1d_real(field_2dreal_ptr, field_2dreal_ptr % missingValueMask1d, real2d_temp)
+                        if (ASSOCIATED( field_2dreal_ptr % missingValueMask2D )) then
+                           call mpas_mask_field2d_mask2d_real(field_2dreal_ptr, field_2dreal_ptr % missingValueMask2d, ownedSize, real2d_temp)
+                           call mpas_log_write('.....masking completed')
+                        else
+                           real2d_temp(:,i:i+ownedSize-1) = field_2dreal_ptr % array(:,1:ownedSize)
+                        endif
                      end if
                      i = i + ownedSize
                      if ( .not. stream % blockWrite ) then
+                        call mpas_log_write('Go to next')
```

Comment:
@akturner and @xylar I don't have any more time to work on this today, but I was able to figure out that the crash happens between this line and the first line of the do-while loop. 

The log is (in part):
```
 Writing out 2d field normalVelocity
 field_2dreal_ptr is normalVelocity
  i,OwnedSize,dimName =            1         150 nEdges
 ... isVar decomposed, assigning real2d_temp
 Masking array normalVelocity
  fieldArray shape=          20         237
  maskedField shape=          20         150
 .....masking completed
 Go to next

Reply:
Great, I'll see if I can make any headway this evening.

Reply:
I got stuck on my time coordinate branch.  It seemed like I had everything working but then it failed catastrophically with a bunch of useless PIO errors.

So once again I didn't get to this -- I had no debugging patience left.  Maybe we can work on it together tomorrow during the work time.

Reply:
I got stuck on my time coordinate branch.  It seemed like I had everything working but then it failed catastrophically with a bunch of useless PIO errors.

So once again I didn't get to this -- I had no debugging patience left.  Maybe we can work on it together tomorrow during the work time.